### PR TITLE
[CI][doc] Add build CI for document changes

### LINKED ISSUE #1716: [Feature][doc] Add build CI in main repository
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

In #1714, we find our doc may crack by some mistake in `mdx` file, to avoid the regression, we should add build CI in the repo and make sure every PR pass the doc build process

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hezean**
## Purpose of this pull request

Closes #1716

When a PR changes the docs, this workflow will try build the website based on the changes, note that this won't affect the existing website repo.

Acknowledgement: `sync.sh` uses several util functions in [incubator-seatunnel-website/tools/build-docs.sh](https://github.com/apache/incubator-seatunnel-website/blob/main/tools/build-docs.sh)

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason: *manually tested*
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: tools/documents/sync.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
```
Comment (CalvinKirs) [inline tools/documents/sync.sh]: Hi you should add the license header

Comment (hezean) [reply tools/documents/sync.sh]: Thanks for reminding, my oversight

**Comment by kezhenxu94** [comment]
@HeZean can you add some random changes in the `docs/**` and see how the workflow works? After that we can revert the changes

**Comment by hezean** [comment]
@kezhenxu94 please check https://github.com/HeZean/incubator-seatunnel/pull/2, you can find the log [here](https://github.com/HeZean/incubator-seatunnel/runs/7552651707?check_suite_focus=true)

**Review by kezhenxu94 — APPROVED** [review]
LGTM

**Pull request merged**